### ISSUE TITLE
Empty WINDOW specifications are not parsed correctly

### ISSUE THREAD

**Issue by matthieucan** [sqlfluff/sqlfluff#1293]
## Expected Behaviour

Empty specification of a window function (`OVER ()`) is parsed correctly as a window function.

## Observed Behaviour

The word `OVER` is interpreted as an alias name, and the parentheses raise a parsing error.

## Steps to Reproduce

```shell
% echo "select sum(1) over()" | python3 -m sqlfluff parse -c -
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'select'
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    function:
[L:  1, P:  8]      |                        function_name:
[L:  1, P:  8]      |                            function_name_identifier:         'sum'
[L:  1, P: 11]      |                        bracketed:
[L:  1, P: 11]      |                            start_bracket:                    '('
[L:  1, P: 12]      |                            expression:
[L:  1, P: 12]      |                                literal:                      '1'
[L:  1, P: 13]      |                            end_bracket:                      ')'
[L:  1, P: 15]      |                    alias_expression:
[L:  1, P: 15]      |                        identifier:                           'over'
[L:  1, P: 19]      |                    unparsable:                               !! Expected: 'Nothing...'
[L:  1, P: 19]      |                        bracketed:
[L:  1, P: 19]      |                            start_bracket:                    '('
[L:  1, P: 20]      |                            end_bracket:                      ')'

==== parsing violations ====
L:   1 | P:  19 |  PRS | Line 1, Position 19: Found unparsable section: '()'
WARNING: Parsing errors found and dialect is set to 'ansi'. Have you configured your dialect?

% echo "select sum(1) over(x)" | python3 -m sqlfluff parse -c -
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'select'
[L:  1, P:  8]      |                select_clause_element:
[L:  1, P:  8]      |                    function:
[L:  1, P:  8]      |                        function_name:
[L:  1, P:  8]      |                            function_name_identifier:         'sum'
[L:  1, P: 11]      |                        bracketed:
[L:  1, P: 11]      |                            start_bracket:                    '('
[L:  1, P: 12]      |                            expression:
[L:  1, P: 12]      |                                literal:                      '1'
[L:  1, P: 13]      |                            end_bracket:                      ')'
[L:  1, P: 15]      |                        over_clause:
[L:  1, P: 15]      |                            keyword:                          'over'
[L:  1, P: 19]      |                            bracketed:
[L:  1, P: 19]      |                                start_bracket:                '('
[L:  1, P: 20]      |                                window_specification:
[L:  1, P: 20]      |                                    identifier:               'x'
[L:  1, P: 21]      |                                end_bracket:                  ')'
```
A `WindowSpecificationSegment` has all its children optional, so I expect it doesn't raise an issue. Not sure where the problem lies.

## Version
`python -m sqlfluff, version 0.6.3` (current `main`)
Python 3.8.10

## Configuration

None

**Comment by tunetheweb** [comment]
Oh I had this true and thought it was bad syntax on our queries. But I see it is allowed so I'd like to see this fixed!

I don't know where the ANSI SQL Standard is, but [BigQuery defines](https://cloud.google.com/bigquery/docs/reference/standard-sql/analytic-function-concepts#def_over_clause) it as this:

```
over_clause:
  { named_window | ( [ window_specification ] ) }
```

That is the `window_specification` is optional, therefore allowing empty brackets. This seems to match other definitions.

So seems to me it's as simple as changing this:

```py
    match_grammar = Sequence(
        "OVER",
        OneOf(
            Ref("SingleIdentifierGrammar"),  # Window name
            Bracketed(
                Ref("WindowSpecificationSegment"),
            ),
        ),
    )
```

to this:

```py
    match_grammar = Sequence(
        "OVER",
        OneOf(
            Ref("SingleIdentifierGrammar"),  # Window name
            Bracketed(
                Ref("WindowSpecificationSegment", optional=True),
            ),
        ),
    )
```

Then it parses correctly. And all other tests still pass too.

I presume it's not really a `WindowSpecificationSegment` it is empty (despite all parts of it being optional), so the better solution is, as I suggested, to set the reference to that in `OverClauseSegment` as optional instead.

Happy for you to submit a PR, or I can do it if you agree with above.

**Comment by tunetheweb** [comment]
I went ahead and submitted #1294

**Comment by matthieucan** [comment]
Oh, awesome! I certainly agree with your solution :)
Didn't think that was the issue since everything within `WindowSpecificationSegment` is optional, but not familiar enough with sqlfluff.
Thank you so much!
